synapse/storage/schema/main/delta/83/05_cross_signing_key_update_grant.sql

```diff
@@ -0,0 +1,15 @@
+/* Copyright 2023 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ALTER TABLE e2e_cross_signing_keys ADD COLUMN updatable_without_uia_before_ms bigint DEFAULT NULL;
```

Comment:
I'd have done it on users themselves but I guess this has the advantage of automatically being ~NULL when a new key gets added to the stream?

Reply:
Yeah, so that the grant to update without UIA is revoked as soon as you upload another key. That sounds like a desirable property to have from a security perspective(?)